sbol_utilities/default_components.py

```diff
@@ -0,0 +1,148 @@
+from typing import List, Tuple
+
+import sbol3
+import tyto
+
+def promoter(name: str, description: str, sequence: str)-> Tuple[sbol3.Component, sbol3.Sequence]:
```

Comment:
- Is this really name, or should it be display_id?
- Can you allow all of the additional arguments to be potentially passed through via kwargs?
- I am guessing that the sequence is assumed to be IUPAC format; that should be documented.
- Should description really be a required argument, and not a kwarg?

Reply:
> Is this really name, or should it be display_id?

In the functions name and display_id are the same. Does that produce a problem? or if I state that on the docstring should be fine? Another solution is to use `display_id = f'{name}_id'`

> Can you allow all of the additional arguments to be potentially passed through via kwargs?

Sure, which arguments do you want to add?

> I am guessing that the sequence is assumed to be IUPAC format; that should be documented.

Should it be on the docstring?

> Should description really be a required argument, and not a kwarg?

Good point, the only required arguments should be the ones required by pySBOL3

Reply:
In terms of parameters: we _have_ to have an identity. That might be provided as a display_id or it might be provided as a full URI. Your functions don't need to distinguish whether they've gotten a full identity or a display_id because in either case it's valid for the constructors and the extension with `_seq`.

For other arguments, you don't need to explicitly provide them: you can use the python kwargs functionality to accept an unbounded set of keyword arguments and to pass them through to the Component constructor. The only arguments that need to be explicit are the ones that get manipulated in these functions.

Reply:
What if instead of name I use identity and then explain in the docstring that function also use that for name? Name could not be included on the function since it is not a required parameter or the function can use identity as name if name is not provided. I would go for the last one.
Do you have a source or an example of how to implement kwargs, I've never used them before.

Reply:
I think it would work to just use identity and allow name to be passed in the kwargs parameter if desired.
The defaulting of an unset name to display_id is already separately proposed as a practice in https://github.com/SynBioDex/pySBOL3/issues/229

On kwargs, here's a tutorial: https://www.geeksforgeeks.org/args-kwargs-python/
You can ask the internet / StackExchange for additional help